internal/agent/handlers/collectLoop.go

```diff
@@ -0,0 +1,35 @@
+package handlers
+
+import (
+	"context"
+	"log/slog"
+	"sync"
+	"time"
+)
+
+type callback = func(s Storage, collectInterval time.Duration) error
+
+func loop(c func(), ctx context.Context, wg *sync.WaitGroup, i time.Duration) {
+	defer wg.Done()
+	timer := time.NewTimer(i + 1)
+	defer timer.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer.C:
+		}
+		c()
+		timer.Reset(i)
```

Comment:
Почему бы не сделать так? Будет более наглядно, как мне кажется.
```suggestion
		case <-timer.C:
			c()
		        timer.Reset(i)
		}
```